mediapipe/graphs/instantmotiontracking/calculators/matrices_manager_calculator.cc

```diff
@@ -211,15 +239,13 @@ ::mediapipe::Status MatricesManagerCalculator::Process(CalculatorContext* cc) {
     // The user transformation data associated with this sticker must be defined
     const float user_rotation_radians = GetUserRotation(user_rotation_data, id);
     const float user_scale_factor = GetUserScaler(user_scaling_data, id);
-    // We can also get the initial scaling factor based on the preset factor for this render id
-    const float render_scale_factor = GetDefaultRenderScale(render_data[render_idx++]);
-    // The final scaling factor (combined user and render preset) can be generated
-    const float scale_factor = user_scale_factor * render_scale_factor;
 
     // A vector representative of a user's sticker rotation transformation can be created
     const Matrix3f user_rotation_submatrix = GenerateUserRotationMatrix(user_rotation_radians);
     // Next, the diagonal representative of the combined scaling data
-    const DiagonalMatrix3f scaling_diagonal = GenerateScalingMatrix(scale_factor);
+    DiagonalMatrix3f scaling_diagonal = GetDefaultRenderScaleDiagonal(render_data[render_idx++], user_scale_factor, gif_aspect_ratio);
+
+    render_idx++;
```

Comment:
There's still a render_idx_++ above; so this seems like it would skip elements, or worse yet, overflow our array bounds.  Maybe remove the one in the brackets to make the logic clearer.